src/openloco/windows/about_music.cpp

```diff
@@ -0,0 +1,187 @@
+
+    namespace widx
+    {
+        enum
+        {
+            frame,
+            title,
+            close,
+            panel,
+            scrollview,
+        };
+    }
+
+    static widget_t _widgets[] = {
+        make_widget({ 0, 0 }, { ww, wh }, widget_type::frame, 0),
+        make_widget({ 1, 1 }, { ww - 2, 13 }, widget_type::caption_25, 0, string_ids::music_acknowledgements_caption),
+        make_widget({ ww - 15, 2 }, { 13, 13 }, widget_type::wt_9, 0, 2321, string_ids::tooltip_close_window),
+        make_widget({ 0, 15 }, { ww, 297 }, widget_type::panel, 1),
+        make_widget({ 4, 18 }, { ww - 8, 289 }, widget_type::scrollview, 1, 2),
+        widget_end(),
+    };
+
+    static window_event_list _events;
+
+    static void on_mouse_up(ui::window* window, widget_index widgetIndex);
+    static void get_scroll_size(ui::window* window, uint32_t scrollIndex, uint16_t* scrollWidth, uint16_t* scrollHeight);
+    static void tooltip(ui::window* window, widget_index widgetIndex);
+    static void draw(ui::window* window, gfx::drawpixelinfo_t* dpi);
+    static void draw_scroll(ui::window* window, gfx::drawpixelinfo_t* dpi, uint32_t scrollIndex);
+
+    // 0x0043B26C
+    void open()
+    {
+        if (windowmgr::bring_to_front(window_type::about_music, 0) != nullptr)
+            return;
+
+        _events.on_mouse_up = on_mouse_up;
+        _events.get_scroll_size = get_scroll_size;
+        _events.tooltip = tooltip;
+        _events.draw = draw;
+        _events.draw_scroll = draw_scroll;
+
+        auto window = windowmgr::create_window_centred(
+            window_type::about_music,
+            ww,
+            wh,
+            0,
+            &_events);
+
+        window->widgets = _widgets;
+        window->enabled_widgets = (1 << widx::close) | (1 << widx::scrollview);
+        window->init_scroll_widgets();
+
+        auto interface = objectmgr::get<interface_skin_object>();
+        window->colours[0] = interface->colour_0B;
+        window->colours[1] = interface->colour_10;
+    }
+
+    static void on_mouse_up(ui::window* window, widget_index widgetIndex)
+    {
+        switch (widgetIndex)
+        {
+            case widx::close:
+                windowmgr::close(window->type);
+                break;
+        }
+    }
+
+    static void get_scroll_size(ui::window* window, uint32_t scrollIndex, uint16_t* scrollWidth, uint16_t* scrollHeight)
+    {
+        *scrollHeight = 1054;
+    }
+
+    static void tooltip(ui::window* window, widget_index widgetIndex)
+    {
+        loco_global<string_id, 0x112C826> common_format_args;
+        *common_format_args = string_ids::tooltip_scroll_credits_list;
+    }
+
+    static void draw(ui::window* window, gfx::drawpixelinfo_t* dpi)
+    {
+        // Draw widgets.
+        window->draw(dpi);
+    }
+
+    static void draw_scroll(ui::window* window, gfx::drawpixelinfo_t* dpi, uint32_t scrollIndex)
+    {
+        static string_id strings_to_draw[] = {
```

Comment:
Move this out of the function

Reply:
Making it local enables more optimisations... Any particular reason why you'd like to move it out?